src/codestudio.js

```diff
@@ -0,0 +1,100 @@
+let redact;
+
+const CODESTUDIO_RE = /^(\[code-studio\s*)(\d+)?-?(\d+)?\]/;
+const CODESTUDIO = "codestudio";
+
+/**
+ * This plugin renders the special "codestudio" syntax used by
+ * CurriculumBuilder for the code studio pullthrough.
+ *
+ * Specifically, given syntax like `[code-studio]`, optionally with range
+ * parameters like `[code-studio 1]` or `[code-studio 3-5]`, it renders a div
+ * with the class "stage-guide" and the optional parameters included as data
+ * attributes.
+ *
+ * @example
+ *
+ *   const parse = require('remark-parse');
+ *   const stringify = require('remark-stringify');
+ *   const unified = require('unified');
+ *
+ *   const source = "Markdown containing [code-studio 1-2] syntax"
+ *   // returns: "<p>Markdown containing <div class="stage-guide" data-start="1" data-end="2"></div> syntax</p>"
+ *   unified().use([
+ *     parse,      // use the standard parser
+ *     codestudio, // use this extension
+ *     stringify,  // output back to markdown
+ *   ]).stringify(source);
+ */
+module.exports = function codestudio() {
+  if (this.Parser) {
+    const Parser = this.Parser;
+    redact = Parser.prototype.options.redact;
+    Parser.prototype.inlineTokenizers[CODESTUDIO] = tokenizeCodeStudio;
+    Parser.prototype.inlineMethods.unshift(CODESTUDIO);
+  }
+};
+
+module.exports.restorationMethods = {
+  [CODESTUDIO]: function(node) {
+    let value = "[code-studio";
+
+    if (node.redactionData.start) {
+      value += " " + node.redactionData.start;
+      if (node.redactionData.end) {
+        value += "-" + node.redactionData.end;
+      }
+    }
+
+    value += "]";
+    return {
+      type: "rawtext",
+      value
+    };
+  }
+};
+
+tokenizeCodeStudio.notInLink = true;
+tokenizeCodeStudio.locator = locateCodeStudio;
+
+function tokenizeCodeStudio(eat, value, silent) {
+  const match = CODESTUDIO_RE.exec(value);
+
+  if (!match) {
+    return;
+  }
+
+  if (silent) {
+    return true;
+  }
+
+  const add = eat(match[0]);
+  const start = match[2];
+  const end = match[3];
+
+  if (redact) {
+    return add({
+      type: "inlineRedaction",
+      redactionType: CODESTUDIO,
+      redactionData: { start, end }
+    });
+  }
+
+  return add({
+    type: "div",
+    data: {
+      hProperties: {
+        className: "stage-guide",
```

Comment:
There's a bug here (repeated in your documentation) unless you intended to update all the JavaScript too: The [original CB implementation](https://github.com/code-dot-org/curriculumbuilder/blob/a06dafa7a6cd36902c1af38e2a5fd0109ed93dbc/curriculumBuilder/codestudio.py#L38) uses the class `stage_guide`, not `stage-guide`.

Reply:
d'oh! Thanks